### PR TITLE
Issue #12229: make Checker names more compact to show profile

### PR DESCRIPTION
**Pull request by romani**
Issue #12229

### CODE REVIEW

**Comment by rnveach** [comment]
Is this taking over https://github.com/checkstyle/checkstyle/pull/12269 ? If so, I would comment or block out the other PR so it is clearer and make sure we don't have unnecessary duplicate efforts?

**Review by rnveach — COMMENTED** [review]
> Checker / analyse (checker-framework-nullness-optional-interning) (pull_request) 

Is it possible and sense to try to make it `Checker / analyse (nullness-optional-interning)`?

**Comment by rnveach** [comment]
Completely unrelated to this PR and can be done in another, but it looks like we can do the same for pitest.

> Pitest / pitest (pitest-coding-2) (pull_request)

**Review by rnveach — APPROVED** [review]
ok, didn't realize what is being displayed is the profile name. I don't know if there is a way to override github workflow display.

**Comment by romani** [comment]
@rnveach , please look now.
Not ideal, but show enough to distinguish

**Comment by romani** [comment]
Screenshot_20221022-072814

**Comment by rnveach** [comment]
another option is to keep profile names as they were, and change checker groovy file to use full profile name.

**Comment by romani** [comment]
lets ask @nrmancuso  for second opinion

File: .github/workflows/checker-framework.yml [hunk 1]
```diff
@@ -7,19 +7,19 @@ on:
...
-          - checker-framework-regex-property-key-compiler-message
-          - checker-framework-purity-value-returns
+          - checker-nullness-optional-interning
+          - checker-methods-resource-fenum
+          - checker-lock-tainting
+          - checker-index
+          - checker-formatter
+          - checker-signature-gui-units-init
+          - checker-regex-property-key-compiler-message
+          - checker-purity-value-returns
```
Comment (nrmancuso) [inline .github/workflows/checker-framework.yml:22]: Let's do this below, in job:
```yml

      - name: Execute checker framework
        env:
          MAVEN_OPTS: "-Xmx4g"
        run: groovy ./.ci/checker-framework.groovy checker-${{ matrix.profile }}

```

then we can just do: `- purity-value-returns`

File: .github/workflows/checker-framework.yml [hunk 2]
```diff
@@ -7,19 +7,19 @@ on:
   pull_request:
 
 jobs:
-  checker-framework:
+  analyse:
```
Comment (nrmancuso) [inline .github/workflows/checker-framework.yml:10]: We can rename to just be `check` to save two more letters for phone-friendly CI diagnostics :)

File: .github/workflows/checker-framework.yml (hunk 1 above)
Comment (romani) [reply .github/workflows/checker-framework.yml]: done

File: .github/workflows/checker-framework.yml (hunk 2 above)
Comment (romani) [reply .github/workflows/checker-framework.yml]: done

**Comment by romani** [comment]
Screenshot_20221022-075515

**Review by rnveach — APPROVED** [review]

**Review by nrmancuso — APPROVED** [review]
Much easier to read and live with.

**Comment by nrmancuso** [comment]
We missed new prefix in other job steps, PR is updated

**Review by nrmancuso — APPROVED** [review]
We needed to rename the files as well, since they contain profile name. It is better to keep all consistent.

**Pull request merged**